src/client/proxy/handlers/deployBlog.ts

```diff
@@ -0,0 +1,166 @@
+    return getFile(CONTRACT_SOURCE_ID);
+}
+
+export async function createSubidentity(subidentity: string) {
+    const [subidentityPart, parentIdentity] = subidentity.split('.');
+    const publicKey = getNetworkPublicKey();
+    const owner = getNetworkAddress();
+    const tx = await ethereum.registerSubIdentity(
+        subidentityPart,
+        parentIdentity,
+        owner,
+        Buffer.from(publicKey, 'hex')
+    );
+    if (tx?.blockHash) {
+        return;
+    }
+    throw 'Could not create subidentity';
+}
+
+export async function ensureSubidentity(subidentity: string) {
+    const id = await ethereum.ownerByIdentity(subidentity);
+    if (id === EMPTY_ADDRESS) {
+        await createSubidentity(subidentity);
+        return true;
+    }
+    return false;
+}
+
+export async function setRootDir(subidentity: string, address: string) {
+    return ethereum.putKeyValue(
+        subidentity,
+        ROOT_DIR_KEY,
+        address,
+        VERSION
+    );
+}
+export async function setRoutes(subidentity: string, address: string) {
+    return ethereum.putKeyValue(
+        subidentity,
+        ZDNS_ROUTES_KEY,
+        address,
+        VERSION
+    );
+}
+
+export async function isUpgreadableBlog(subidentity: string, abiAddress: string) {
+    const [zdnsRoutes, blogContractAddress] = await getManyKeys(subidentity, [
+        'zdns/routes',
+        `${CONTRACT_PREFIX}/address/Blog`
+    ]);
+    if (zdnsRoutes) {
+        const routes = await getFile(zdnsRoutes);
+        routes.includes(`"/blog": "index.html"`);
+        if (blogContractAddress && await verifyBlogABI(abiAddress)) {
+            return true;
+        } else {
+            throw 'A website is already deployed on this address. Please, choose another subidentity';
+        }
+    }
+}
+
+export function needsToUpdateContract(abiAddress: string) {
+    return abiAddress !== CURRENT_ABI_ID;
+}
+export async function deployBlog(subidentity: string) {
+    const isNewIdentity = await ensureSubidentity(subidentity);
+    const [abiAddress] = await getManyKeys(subidentity, [`${CONTRACT_PREFIX}/abi/Blog`]);
+    if (!isNewIdentity && !await isUpgreadableBlog(subidentity, abiAddress)) {
+        throw 'Is not an upgreadable blog';
+    }
+    const [rootDir, routes] = await getManyKeys(
+        subidentity, [
+            ROOT_DIR_KEY,
+            ZDNS_ROUTES_KEY
+        ]
+    );
+    const blogContract = await getBlogContractCode();
+    if (needsToUpdateContract(abiAddress)) {
+        await deployUpgradableContracts({
+            contracts: [{file: Buffer.from(blogContract), name: 'Blog'}],
+            version: VERSION,
+            target: subidentity,
+            forceDeployProxy: isNewIdentity,
+            dependencies: BLOG_DEPENDENCIES
+        });
+    }
+    if (rootDir !== ROOT_DIR_ID) {
+        await setRootDir(subidentity, ROOT_DIR_ID);
+    }
+    if (routes !== ROUTES_FILE_ID) {
+        await setRoutes(subidentity, ROUTES_FILE_ID);
```

Comment:
It would be much more readable if you put the code of this function here, as long as this function is just an alias of another one, and actually is not called anywhere else.